lib/remote_node.go

```diff
@@ -0,0 +1,468 @@
+package lib
+
+import (
+	"encoding/binary"
+	"fmt"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/deso-protocol/core/bls"
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/sha3"
+	"net"
+	"time"
+)
+
+type RemoteNodeConnectionType int
+
+const (
+	RemoteNodeConnectionType_Unknown  RemoteNodeConnectionType = 0
+	RemoteNodeConnectionType_Outbound RemoteNodeConnectionType = 1
+	RemoteNodeConnectionType_Inboound RemoteNodeConnectionType = 2
+)
+
+type RemoteNodeConnectionStatus int
+
+const (
+	RemoteNodeConnectionStatus_NotConnected RemoteNodeConnectionStatus = 0
+	RemoteNodeConnectionStatus_Connected    RemoteNodeConnectionStatus = 1
+	RemoteNodeConnectionStatus_Attempted    RemoteNodeConnectionStatus = 2
+	RemoteNodeConnectionStatus_Terminated   RemoteNodeConnectionStatus = 3
+)
+
+type RemoteNode struct {
+	// Should we have ID here?
```

Comment:
Is the ID you're referring to here different than the `id` field on line 35?

Reply:
I've addressed this now. The RemoteNode now has a singular Id, and the OutboundConnectionAttempt & Peer will inherit this Id.